omnigibson/controllers/joint_controller.py

```diff
@@ -204,7 +215,11 @@ def compute_control(self, goal_dict, control_dict):
                 # Run impedance controller -- effort = pos_err * kp + vel_err * kd
                 position_error = target - base_value
                 vel_pos_error = -control_dict[f"joint_velocity"][self.dof_idx]
-                u = position_error * self.pos_kp + vel_pos_error * self.pos_kd
+
+                # Update integral error
+                self._integral_error += position_error * (1.0 / self.control_freq)  # Integrate error over time
+                self._integral_error = self._integral_error.clip(-self.max_integral_error, self.max_integral_error)
+                u = position_error * self.pos_kp + vel_pos_error * self.pos_kd + self._integral_error * self.pos_ki
```

Comment:
@cremebrule @cgokmen should also review this. Any implication? Did all the robot tests still pass? Does robot_control_example still work?